.github/workflows/test.yml

```diff
@@ -46,11 +46,28 @@ jobs:
           export PATH="${PATH}:${GOPATH}/bin"
           go test -v ./... -race
       - name: Run Upgrade Tests
+        id: upgrade_tests
+        continue-on-error: true
```

Comment:
why do we want to mark failing tests as passed?

Reply:
comments are extremely easy to miss. Failed tests less so

Reply:
You're right, looking at it I don't like this approach either.
The original idea was to separate go unit testing from upgrade testing, as they serve different purposes, so we can know what exactly is failing without having to open the gh action. Instead of the current workaround I'll be splitting it into a different workflow, so we don't see continuous "go test failing" when it's an upgrade test.